build_dependency_tree.py

```diff
@@ -3,76 +3,202 @@
+                if '/*' in line:
                     b_is_block_comment = True
                     continue
-                elif line.split() == '//':
+                elif line.replace(' ', '')[0:2] == '//':
+                    continue
+                elif not line:  # empty line
                     continue
-                elif line.split()[0] == '#include':
-                    # match either the pattern `<...>` or `"..."`
-                    # re.findall will return a list of tuples `[(pattern 1, pattern 2), ...]`
-                    re_match = re.findall(r"<(.+?.)>|\"(.+?.)\"", line.split()[1])[0]
+                elif line.replace(' ', '')[0:8] == '#include' or \
+                        line.replace(' ', '')[0:9] == '%:include':
+                    # we always expect the included file to be given second
+                    # (required by standards)
+                    re_match = include_statement_pattern.match(line.split()[1])
                     # We only expect one entry in the list
-                    if not re_match[0]: # first matching group is empty
+                    if not re_match:  # first matching group is empty
+                        print(f'No header files found in line: {line}')
                         continue
+                    elif not re_match.group(1):
+                        # Check if match with `<...>` is None
+                        print(f"""Header file found
+                                {str(file.absolute())}:{re_match.group(2)}""")
+                        include_statements.append(
+                                extract_header_file_name(
+                                    re_match.group(2)))
                     else:
-                        include_statements.append(re_match[0])
-                else:
-                    continue
+                        # since we have a match and first group is non-empty we
+                        # apend
+                        print(f"""Header file found
+                                {str(file.absolute())}:{re_match.group(1)}""")
+                        include_statements.append(
+                                extract_header_file_name(
+                                    re_match.group(1)))
         return include_statements
+
+
+def output_dependency_tree_to_dot_file(
+        dep_tree: Dict[List[Union[Path, str]], str],
+        output_name: Optional[Union[Path, str]]):
+    """
+    Goes through dictionary of files gathered from recursive search of
+    directory and prints all the nodes it will be attached to in file
+    named output_name.
+    Output is formated to be compatible with the DOT command line program:
+    https://graphviz.org/doc/info/lang.html
+
+    :param dep_tree:    dictionary with files in project as keys and list
+                        of their dependencies as value
+    :param output_name: name of output file or path to output file
+    """
+    if not isinstance(dep_tree, dict):
```

Comment:
Are expecting to have anything other than a dict passed in? if this is supposed to be of value `None`, python will through a `RunTimeError` as our typing declares this has to be Dict. or if anything that's not a dict gets passed in for that matter

Reply:
I will remove this in the refactor, since "ideally" nobody will edit source code and this function only gets called from our code anyway.
But for now lets agree to leave it in.